cli/hclfmt_test.go

```diff
@@ -191,3 +191,84 @@ func TestHCLFmtCheckErrors(t *testing.T) {
 		})
 	}
 }
+
+func TestHCLFmtFile(t *testing.T) {
```

Comment:
Is there a reason you created new fixtures for this instead of reusing the existing ones (e.g., https://github.com/gruntwork-io/terragrunt/tree/master/test/fixture-hclfmt-check)?

Reply:
Also, these are nice tests to be comprehensive, but since we already have comprehensive testing of the functionality of `hclfmt`, it kind of overlaps with the other tests. The real test we need here is one that checks that the `--terragrunt-hclfmt-file` option only touches the file and not recursively hit all the subdirectories.

So I think it is sufficient to have a single test where `HclFile` is set to `fixture-hclfmt-check/a/terragrunt.hcl`, and the root (or `TerragruntConfigPath`) is set to `fixture-hclfmt-check/terragrunt.hcl`, and checks that only the set file was formatted the same as `fixture-hclfmt-check/expected.hcl`. It should check to make sure that `fixture-hclfmt-check/terragrunt.hcl` and `fixture-hclfmt-check/b/c/terragrunt.hcl` were untouched.

What do you think?

Reply:
> Is there a reason you created new fixtures for this instead of reusing the existing ones (e.g., https://github.com/gruntwork-io/terragrunt/tree/master/test/fixture-hclfmt-check)?

i definitely could've clarified before setting up the fixtures but I was under the impression we wanted very clear separation of concerns especially since there are redundancies between the hclfmt and hclfmt-check-errors fixtures.

In any case, I think your testing makes more sense and I'll give it a go!

Reply:
i just made the changes and targeted the test at `fixture-hclfmt` instead because `fixture-hclfmt-check` has hcl files that are all identical. otherwise, I did as suggested! it runs the formatting on the hcl file in a and then checks the other two hcl files with the contents of the original file at `fixture-hclfmt/terragrunt.hcl`